posts/2022-05-28-liberty-deep-dive.adoc

```diff
@@ -0,0 +1,45 @@
+---
+layout: post
+title: "Introducing a hands-on technical deep dive on Liberty!"
+# Do NOT change the categories section
+categories: blog
+author_picture: https://avatars3.githubusercontent.com/ryan-storey
+author_github: https://github.com/ryan-storey
+seo-title: Introducing a hands-on technical deep dive on Liberty! - OpenLiberty.io
+seo-description: Introducing an end-to-end tutorial on cloud-native Java application development and deployment using Jakarta EE, MicroProfile and Liberty. The tutorial covers building RESTful microservice, documenting your REST API, externalizing configuration, persisting data to a database, securing application using JSON Web Tokens (JWT), adding health checks, providing metrics, testing using Testcontainers, building container images, and deploying to Kubernetes using Operator.
+blog_description: "Introducing an end-to-end tutorial on cloud-native Java application development and deployment using Jakarta EE, MicroProfile and Liberty"
+open-graph-image: https://openliberty.io/img/twitter_card.jpg
+---
+= Introducing a hands-on technical deep dive on Liberty!
+Ryan Storey <https://github.com/ryan-storey>
+:imagesdir: /
+:url-prefix:
+:url-about: /
+//Blank line here is necessary before starting the body of the post.
+
+We are excited to announce the release of our newest tutorial, A Technical Deep Dive on Liberty.
+
+A Technical Deep Dive on Liberty is an end-to-end tutorial that walks you through how to develop and deploy cloud-native Java applications in Jakarta EE and MicroProfile. The tutorial is aimed at Java developers, from Liberty novices to those more experienced with writing applications to run on Liberty. This tutorial is aimed to help developers, from Liberty novices to those more experienced, develop, build, test, and deploy your applications on Liberty running in Kubernetes and clouds.
+
+In A Technical Deep Dive on Liberty, you will learn how to:
+
+* build a RESTful microservice on Liberty with Jakarta EE and MicroProfile 
+* document your REST API 
+* externalize your configuration
+* persist data to a database 
+* secure your application using JSON Web Tokens (JWT) 
+* add health checks 
+* provide metrics 
+* build a container image for your Java microservice
+* deploy the container image to Kubernetes using the Open Liberty Operator
+* test your applications, and more. 
```

Comment:
A suggestion (@yeekangc - what do you think? depends on whether you want to publish this week more)...could the benefits of some of these be explained a little?
eg "easily document your REST API so that other developers can use it"; "externalize your configuration so that it can be easily, and automatically, be changed throughout the development lifecycle"; "add health checks to automatically verify whether your microservice is available or not"; "provide metrics to monitor the activity of your microservice".

Can probably drop the ", and more." bit from the last bullet. There's quite a lot listed already!

Reply:
Good point!  Great if we can do this and it will be helpful to the readers too.

Reply:
* "build a RESTful microservice on Liberty with Jakarta EE and MicroProfile and make it available for requests from other applications"
* "easily document the REST API so that other developers can use it to make requests to your microservice"
* "externalize the configuration of your microservice so that it can be easily, and automatically, be changed throughout the development lifecycle"
* "persist data to a database so that your microservice can look up or store information from requests"
* "secure your application using JSON Web Tokens (JTW) so that only authenticated and authorized applications can request information from your microservice"
* "add health checks to automatically verify whether your microservice is available or not"
* "provide metrics to monitor the activity of your microservice"
* "build a container image

Reply:
I feel we don't need the additional qualifications for "build a RESTful microservice on Liberty with Jakarta EE and MicroProfile" and ""persist data to a database" since they are or should be obvious to a person familiar with the art?

The one for JWT is good though.